server/src/http/routes/psFormation.js

```diff
@@ -60,15 +78,27 @@ module.exports = ({ catalogue, tableCorrespondance }) => {
   );
 
   /**
-   * Update psFormation with reconciliated data
+   * Update one establishment type in matching_mna_etablissement array
    */
-
   router.put(
-    "/",
+    "/etablissement",
     tryCatch(async (req, res) => {
-      const data = req.body;
-      const response = await PsFormation.findByIdAndUpdate(data._id, { ...data }, { new: true });
-      res.json(response);
+      const { formation_id, etablissement_id, type } = req.body;
+      const update = await PsFormation.updateOne(
+        { _id: formation_id },
+        { $set: { "matching_mna_etablissement.$[elem].type": type } },
+        { arrayFilters: [{ "elem._id": new mongoose.Types.ObjectId(etablissement_id) }] }
+      );
+      if (update) {
+        if (update.nModified === 1) {
+          const response = await PsFormation.findById({ _id: formation_id });
+          res.json(response);
+        } else {
+          res.json(update);
+        }
+      } else {
+        res.status(400).json(update);
```

Comment:
Here I guess `update` is kind of empty, right ?

Reply:
oh yes ! thanks